### PR TITLE
[osascript] refactor deprecated `util.isArray`

### PR DESCRIPTION
**Pull request by hirbod**
# Why
> (node:4385) [DEP0044] DeprecationWarning: The `util.isArray` API is deprecated. Please use `Array.isArray()` instead.

This is done in preparation for Node 22 (Array.isArray is available in Node 10+)

# How
Changed util.isArray to Array.isArray. I did not remove the util import, as other util functions are still in use.

# Test Plan

# Checklist

- [X] Documentation is up to date to reflect these changes (eg: https://docs.expo.dev and README.md).
- [X] Conforms with the [Documentation Writing Style Guide](https://github.com/expo/expo/blob/main/guides/Expo%20Documentation%20Writing%20Style%20Guide.md)
- [X] This diff will work correctly for `npx expo prebuild` & EAS Build (eg: updated a module plugin).

### CODE REVIEW

**Comment by Simek** [comment]
Hey @hirbod, can you also rebuild the package files?

**Comment by hirbod** [comment]
@Simek done

File: packages/@expo/osascript/CHANGELOG.md [hunk 1]
```diff
@@ -8,6 +8,8 @@
 
 ### 🐛 Bug fixes
 
+- Fix util.isArray is deprecated ([#28927](https://github.com/expo/expo/pull/28927) by [@hirbod](https://github.com/hirbod))
```
Comment (Simek) [inline packages/@expo/osascript/CHANGELOG.md:11]:
```suggestion
- Replace deprecated `util.isArray` helper. ([#28927](https://github.com/expo/expo/pull/28927) by [@hirbod](https://github.com/hirbod))
```

**Review by Simek — APPROVED** [review]

**Comment by byCedric** [comment]
Released as `@expo/osascript@2.1.3`

**Pull request merged**